src/Sulu/Bundle/MarkupBundle/Markup/HtmlTagExtractor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MarkupBundle\Markup;
+
+/**
+ * Extracts tags from html content.
+ */
+class HtmlTagExtractor implements TagExtractorInterface
+{
+    const COUNT_REGEX = '/<%1$s:[a-z]+/';
+    const ATTRIBUTE_REGEX = '/(?<name>\b[\w-]+\b)\s*=\s*"(?<value>[^"]*)"/';
+    const CONTENT_REGEX = '/(?:>(?<content>[^<]*)<)/';
+    const TAG_REGEX = '/(?<tag><%1$s:(?<name>[a-z]+)((?!>|\/>).)*(?:\/>|>((?!<\/%1$s:\2*>).)*<\/%1$s:\2*>))/';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count($html, $namespace)
+    {
+         return preg_match_all(sprintf(self::COUNT_REGEX, $namespace), $html, $matches);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extract($html, $namespace)
+    {
+        if (!preg_match_all(sprintf(self::TAG_REGEX, $namespace), $html, $matches)) {
+            return [];
+        }
+
+        $sortedTags = [];
+        for ($i = 0, $length = count($matches['name']); $i < $length; ++$i) {
+            $tag = $matches['tag'][$i];
+            $name = $matches['name'][$i];
+            if (!array_key_exists($name, $sortedTags)) {
+                $sortedTags[$name] = [];
+            }
+
+            $sortedTags[$name][$tag] = $this->getAttributes($tag);
+        }
+
+        return $sortedTags;
+    }
+
+    /**
+     * Returns attributes of given html-tag.
+     *
+     * @param string $tag
+     *
+     * @return array
+     */
+    private function getAttributes($tag)
```

Comment:
And I am not a big fan of simply making everything `protected`, because it limits us and forces us to also populate internal logic, which doesn't necessarily need to be the case. Even if we make this method protected you have to replace it completely in your own implementation. And whatever we are doing in here, you could also have an impact on the result of the `extract` method if you only override this one using the `Decorator` pattern or simply inheriting from it. If you use the `Decorator` you are even not tied to our specific implementation, which is another nice thing 😃